Assets.Scripts.Core.Audio/AudioLayerUnity.cs

```diff
@@ -198,12 +199,12 @@ private void OnFinishLoading()
 
 		private void OnAudioEnd()
```

Comment:
Took me a bit to see what was going on with the changes in this file - it's a little confusing, but if it gets the job done that's fine.

Reply:
(confusing especially with the swapping around and local variable copy of the callback)

Reply:
Ahh yeah the issue was that if the `finishCallback` tried to enqueue a new piece of audio it would get stopped immediately.  Also stopping the audio early didn't clear the finishCallback before, so I changed it so that it does.  I checked and none of the other code actually used the finish callback system before, but we should probably check future releases to make sure they don't use it and rely on the old (weird, not sure how you could rely on it) behavior.

Reply:
Ahh yeah the issue was that if the `finishCallback` tried to enqueue a new piece of audio it would get stopped immediately.  Also stopping the audio early didn't clear the finishCallback before, so I changed it so that it does.  I checked and none of the other code actually used the finish callback system before, but we should probably check future releases to make sure they don't use it and rely on the old (weird, not sure how you could rely on it) behavior.